src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -286,3 +286,17 @@ void Terminal::ColorSelection(const COORD, const COORD, const TextAttribute)
 {
     THROW_HR(E_NOTIMPL);
 }
+
+// Method Description:
+// - If the clicked text is a hyperlink, open it
+// Arguments:
+// - The position of the clicked text
+void Terminal::OpenHyperlink(const COORD position)
+{
+    auto attr = _buffer->GetCellDataAt(_ConvertToBufferCell(position))->TextAttr();
+    if (attr.IsHyperlink())
+    {
+        auto uri = _buffer->GetHyperlinkUriFromId(attr.GetHyperlinkId());
+        ShellExecute(nullptr, L"open", uri.c_str(), nullptr, nullptr, SW_SHOWNORMAL);

```

Comment:
Aren't there security issues with just opening any uri we receive? Shouldn't we be whitelisting particular schemes we consider safe?

Reply:
We check for http or https schemes now!